src/os/bluestore/BlueStore.h

```diff
@@ -1649,6 +1671,31 @@ class BlueStore : public ObjectStore,
       modified_objects.insert(o);
     }
 
+    void note_new_object(OnodeRef &o) {
+      ceph_assert(onode_to_offset_map.count(o) == 0);
+      onode_to_offset_map[o] = {o->get_ondisk_offset()};
+    }
+
+    void note_updated_object(OnodeRef &o, int64_t prev_offset) {
+      int64_t new_offset = o->get_ondisk_offset();
+      auto it = onode_to_offset_map.find(o);
```

Comment:
same trick with insert/emplace is available here

Reply:
Done.